brax/physics/colliders.py

```diff
@@ -35,10 +35,12 @@ class Collidable:
   corner collidables.
   """
 
-  def __init__(self, collidables: List[config_pb2.Body], body: bodies.Body):
+  def __init__(self, collidables: List[config_pb2.Body], body: bodies.Body, config: config_pb2.Config):
```

Comment:
Rather than adding configs to the signatures to fetch global friction, let's just pre-process the config to insert materials with the globals where missing.  We already do this kind of config pre-processing here:

https://github.com/google/brax/blob/main/brax/physics/base.py#L147

So should be straightforward to add more pre-processing.  Then you can just depend on colliders always having materials in the code.